sdk/python/kfp/components/_python_op.py

```diff
@@ -694,6 +695,118 @@ def add_multiply_two_numbers(a: float, b: float) -> NamedTuple('DummyName', [('s
+    You can specify the type of the consumed/produced data by specifying the type argument to `InputPath` and `OutputPath`. The type can be a python type or an arbitrary type name string. `OutputPath('CatBoostModel')` means that the function states that the data it has written to a file has type 'CatBoostModel'. `InputPath('CatBoostModel')` means that the function states that it expect the data it reads from a file to have type 'CatBoostModel'. When the pipeline author connects inputs to outputs the system checks whether the types match.
+    Every kind of data can be consumed as a file input. Conversely, bigger data should not be consumed by value as all value inputs pass through the command line.
+
+    Example of a component function declaring file input and output::
+
+        def catboost_train_classifier(
+            training_data_path: InputPath('CSV'),            # Path to input data file of type "CSV"
+            trained_model_path: OutputPath('CatBoostModel'), # Path to output data file of type "CatBoostModel"
+            number_of_trees: int = 100,                      # Small output of type "Integer"
+        ) -> NamedTuple('Outputs', [
+            ('Accuracy', float),  # Small output of type "Float"
+            ('Precision', float), # Small output of type "Float"
+            ('JobUri', 'URI'),    # Small output of type "URI"
+        ]):
+            """Trains CatBoost classification model"""
+            ...
+
+            return (accuracy, precision, recall)
+
+
+    Args:
+        func: The python function to convert
+        base_image: Optional. Specify a custom Docker container image to use in the component. For lightweight components, the image needs to have python 3.5+. Default is the python image corresponding to the current python environment.
+        output_component_file: Optional. Write a component definition to a local file. The produced component file can be loaded back by calling `load_component_from_file` or `load_component_from_uri`.
+        packages_to_install: Optional. List of [versioned] python packages to pip install before executing the user function.
+
+    Returns:
+        A factory function with a strongly-typed signature taken from the python function.
+        Once called with the required arguments, the factory constructs a task instance that can run the original function in a container.
+    '''
+
+    component_spec = _func_to_component_spec(
+        func=func,
+        base_image=base_image,
+        packages_to_install=packages_to_install,
+    )
+
+    if output_component_file:
+        component_spec.save(output_component_file)
+
+    return _create_task_factory_from_component_spec(component_spec)
```

Comment:
Provided the similarity b/w the functions of this one and `func_to_container_op`, do we have a plan for deprecating one or another to prevent duplicated logic and multiple sources of truth?

Reply:
In the future, `create_component_from_func` should supersede `func_to_container_op`.
The `func_to_container_op` would look strange in systems that don't have `ContainerOp` and `create_component_from_func` fits nicely with `create_component_from_airflow_op` and  `create_graph_component_from_func`.

P.S.
It should probably be called like that from the beginning, but I had some reservations.
`_func_to_component_spec` returns `ComponentSpec`, so the name is correct. The `_create_task_factory_from_component_spec` function also has correct name. But how should I name the "func_to_container_op" function? The correct name would have been `func_to_component_task_factory` or `create_task_factory_from_func`, but that would be too long and complicated. I've presented couple of names to the team and they liked the `func_to_container_op` name.